src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAO.scala

```diff
@@ -123,6 +130,47 @@ class ElasticSearchDAO(client: TransportClient, indexName: String, ontologyDAO:
     populateSuggestions(client, indexName, field, text)
   }
 
+  override def statistics: LogitMetric = {
+    if (FireCloudConfig.Metrics.libraryNamespaces.isEmpty)
+      throw new FireCloudException("no namespaces defined for ElasticSearchDAO.statistics()")
+
+    val AGGKEY = "sumSamples"
+
+    // build query: any namespace defined in our conf file
+    val namespaceFilter = boolQuery()
+    FireCloudConfig.Metrics.libraryNamespaces.map { ns =>
+      namespaceFilter.should(termQuery("namespace.keyword", ns))
+    }
+    val namespaceQuery = boolQuery().filter(namespaceFilter)
+
+    // build aggregation: sum of the numSubjects field
+    val sum = AggregationBuilders.sum(AGGKEY).field("library:numSubjects")
+
+    // build the search, using query and aggregation. Set size to 0; we don't care about search results, only
+    // the aggregation results.
+    val search = client.prepareSearch(indexName).addAggregation(sum).setQuery(namespaceQuery).setSize(0)
+
+    // execute search
+    val statsTry = Try(executeESRequest[SearchRequest, SearchResponse, SearchRequestBuilder](search))
+
+    // extract the sum from the results, safely
+    val safeNumber = statsTry match {
+      case Success(stats) =>
+        val sum = stats.getAggregations.get[Sum](AGGKEY)
+        if (sum != null) {// eww, Java nulls
+          sum.getValue
+        } else {
+          logger.info(s"statistics query did not find $AGGKEY in aggregation results!")
+          0
+        }
+      case Failure(ex) =>
+        logger.warn(s"statistics query failed: ${ex.getMessage}")
+        0
+    }
+
+    NumSubjects(safeNumber.toInt)
```

Comment:
If we move `.toInt` to after `.getValue` then we don't have to convert the zeros to int also